### PR TITLE
161 Datablocks: refactor tarfile creation

### PR DESCRIPTION
**Pull request by phwissmann**
Improves the tarfile creation and add tests for the corner cases.

As discussed, this currently only considers flat folder structures to estimate the size of the final tarfile

### CODE REVIEW

**Review by pulsargranular — COMMENTED** [review]
Creating our own custom-made tarfiles feels a bit odd. Since your implementation does not solve the (possible) problem of very big files, I would suggest to use RAR for this. It supports multivolume archives which would perfectly fit our purpose. It can even repair and recover damaged archives.

**Comment by phwissmann** [comment]
> I would propose a quite different approach: the ingestor should do the checksum and zipping and upload (using several streams) to S3 storage. This can be achieved without having to additionally store the zipped file on the client disk. The archiver process should then split the zip file into smaller chunks, according to the needs of the archiving system.

I don't think this is feasible at this point in time.

File: backend/archiver/utils/datablocks.py [hunk 1]
```diff
@@ -42,20 +42,51 @@ class TarInfo:
     path: Path
 
 
+def partition_files_flat(folder: Path, target_size_bytes: int) -> Generator[List[Path], None, None]:
```
[the remark this answers is not in the document]
Comment (phwissmann) [reply backend/archiver/utils/datablocks.py:45]: > passed around all the time
that is strong wording ^^

I see this function and the generator below taking it. I didn't want to couple the implementation to prefect where the value is coming from.
